convert.go

```diff
@@ -0,0 +1,147 @@
+// Package flatfs is a Datastore implementation that stores all
+// objects in a two-level directory structure in the local file
+// system, regardless of the hierarchy of the keys.
+package flatfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+)
+
+func Convert(oldPath string, getOldDir ShardFunc, newPath string, getNewDir ShardFunc, out io.Writer) error {
+	old, err := New(oldPath, getOldDir, false)
+	if err != nil {
+		return err
+	}
+	new, err := New(newPath, getNewDir, false)
+	if err != nil {
+		return err
+	}
+
+	if out != nil {
+		fmt.Fprintf(out, "Getting Keys...\n")
+	}
+	res, err := old.Query(query.Query{KeysOnly: true})
+	if err != nil {
+		return err
+	}
+	entries, err := res.Rest()
```

Comment:
@whyrusleeping I was copying the method you used in https://github.com/ipfs/fs-repo-migrations/blob/master/ipfs-3-to-4/migration/migration.go#L231.

I gather the keys first so I can get an accurate count for the progress bar.

And you are right it will take a good deal of memory, but it it didn't seam to cause anyone a problem the last time a conversion was made.